deploy.sh

```diff
@@ -11,31 +9,124 @@ export TAG=1.7-SNAPSHOT
+    do
+        sleep 2 ## sleep 2 * $count 150 == 300 sec timeout
+        echo -n "."
+        (( count = count + 1 ))
+        if [ $count -gt 150 ]
+        then
+            echo " failed."
+            # Find any nodes that are in error, and output their logs
+            for pod in $(oc get pods --no-headers | grep -v Running | cut -f1 -d" ")
+            do
+                echo Logs for $pod:
+                oc logs --all-containers $pod || :
+            done
+            exit 1
+        fi
+    done
+
+    echo " Pods up!"
+}
+
 
-oc login https://$OPENSHIFT_SERVER --token=$TOKEN
+testDeployment() {
+    echo "Try this deployment using:"
+    echo "export OPENSHIFT_SERVER=centralpark2.lightbend.com"
+    echo "export OPENSHIFT_PROJECT=lagom-scala-openshift-smoketests"
+    echo 'curl -H "Host: my-lagom-openshift-smoketests.example.org"  https://$OPENSHIFT_PROJECT.$OPENSHIFT_SERVER/proxy/rest-hello/alice'
 
-oc delete project $OPENSHIFT_PROJECT
-# while the project is deleted, we build the docker imges
-sbt clean docker:publishLocal
-oc new-project    $OPENSHIFT_PROJECT
+    count=0
+    echo -n "Waiting for smoketest to be form cluster and serve traffic."
+    while [ "$(curl -H "Host: my-lagom-openshift-smoketests.example.org"  https://$OPENSHIFT_PROJECT.$OPENSHIFT_SERVER/proxy/rest-hello/alice | grep "Hi alice" | wc -l )" -ne 1 ]
+    do
+        sleep 2 ## sleep 2 * $count 150 == 300 sec timeout
+        echo -n "."
+        (( count = count + 1 ))
+        if [ $count -gt 150 ]
+        then
+            echo " Pods are up but service is failing."
+            # Find any nodes that are in error, and output their logs
+            for pod in $(oc get pods --no-headers | grep -v Running | cut -f1 -d" ")
+            do
+                echo Logs for $pod:
+                oc logs --all-containers $pod || :
+            done
+            exit 1
+        fi
+    done
 
-docker login -p $TOKEN -u unused $DOCKER_REGISTRY_SERVER
+    echo " Service up and running."
 
-echo "Tagging image: $IMAGE_HELLO:$TAG $DOCKER_REGISTRY/$IMAGE_HELLO:$TAG"
-echo "  - $IMAGE_HELLO:$TAG"
-echo "  - $DOCKER_REGISTRY/$IMAGE_HELLO:$TAG"
-docker tag $IMAGE_HELLO:$TAG $DOCKER_REGISTRY/$IMAGE_HELLO:$TAG
 
-echo "Tagging image: $IMAGE_HELLO_PROXY:$TAG $DOCKER_REGISTRY/$IMAGE_HELLO_PROXY:$TAG"
-echo "  - $IMAGE_HELLO_PROXY:$TAG"
-echo "  - $DOCKER_REGISTRY/$IMAGE_HELLO_PROXY:$TAG"
-docker tag $IMAGE_HELLO_PROXY:$TAG $DOCKER_REGISTRY/$IMAGE_HELLO_PROXY:$TAG
+}
 
-docker push $DOCKER_REGISTRY/$IMAGE_HELLO:$TAG
-docker push $DOCKER_REGISTRY/$IMAGE_HELLO_PROXY:$TAG
 
-kustomize build deployment/overlays/openshift | oc apply -f -
 
-echo "Try this deployment using:"
-echo 'curl -H "Host: my-lagom-openshift-smoketests.example.org"  https://$OPENSHIFT_PROJECT.$OPENSHIFT_SERVER/proxy/rest-hello/alice'
\ No newline at end of file
+login
+deleteProject
+buildImages
+createProject
+pushImages
+deploy
+waitForApp
+testDeployment
```

Comment:
should we call `deleteProject` at the end?

Reply:
I prefer calling delete at the beginning to prevent running on a dirty environment. It is also helpful to not delete the project on the last step because: 1) we have a sample app available at all times for free and b) allows inspecting the running app if an issue arises.

A `delete` at the beginning is required anyway because we're never sure the target environment will be clean.